CASMINST-5942: Modify NCN image customization procedure to be more flexible

mharding-hpe opened the pull request:
Backport of https://github.com/Cray-HPE/docs-csm/pull/3265

(note that this is actually a smaller set of changes for 1.4 and main, because in those releases the S3 path will always be an IMS image ID -- these changes just carry forward some of the linting / quality of life improvements from the original PR)

mharding-hpe commented:
/backport main

don-bahls-hpe commented:
For CSM-1.4, it would be good to mention  that there are IUF workflows for building images.   I don't recall us needing to look up the artifact IDs, we just leveraged sat bootprep.   That said it is probably worthwhile to document the underlying process still.

haasken-hpe commented on operations/configuration_management/Management_Node_Image_Customization.md:
```diff
@@ -17,29 +18,98 @@ with minor command modifications.
+         > The command used in this substep is extracting the location of the NCN image from the `metal.server` boot parameter for the
+         > NCN in BSS. For more information on that parameter, see [`metal.server` boot parameter](../../background/ncn_kernel.md#metalserver).
+
+1. (`ncn-mw#`) Obtain the NCN image's associated artifacts (SquashFS, kernel, and `initrd`).
```
I don't understand why this step is added here. These files are getting downloaded and then not used anywhere else in this documentation, as far as I can tell.

This looks like part of the old process that was used to download the image artifacts from S3 in order to then create an IMS image record, re-upload the image artifacts to the appropriate path in IMS, create the IMS image manifest that references those artifacts, and then linking that manifest to the IMS image record. In CSM 1.4, this is already done for us as part of the CSM installation.

haasken-hpe reviewed (approved):
The changes here look good, apart from the seemingly unnecessary step to download image artifacts from S3.

I should mention that I think this entire page needs to be simplified to talk about using `sat bootprep` to perform this process. It also needs to somehow reference how this procedure is already performed as part of the prepare-images stage of the IUF. I have a Jira [CASMINST-5764](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-5764) to work on this, and I discussed yesterday with @deantroe and @leliasen-hpe.

mharding-hpe commented:
> The changes here look good, apart from the seemingly unnecessary step to download image artifacts from S3.
> 
> I should mention that I think this entire page needs to be simplified to talk about using `sat bootprep` to perform this process. It also needs to somehow reference how this procedure is already performed as part of the prepare-images stage of the IUF. I have a Jira [CASMINST-5764](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-5764) to work on this, and I discussed yesterday with @deantroe and @leliasen-hpe.

Agreed.

I removed the now-unnecessary step from this PR (and the main PR), and also removed the references to this procedure from the IMS upload procedure (again, in this PR and the main PR). I think these PRs are now good to go. As you noted, further refactoring work is needed for 1.4+, but we can deal with that using your other ticket.
